hibernate-core/src/main/java/org/hibernate/query/sqm/produce/function/ArgumentTypesValidator.java

```diff
@@ -253,6 +253,7 @@ private static boolean isCompatible(FunctionParameterType type, JdbcType jdbcTyp
 			case IMPLICIT_JSON -> jdbcType.isImplicitJson();
 			case XML -> jdbcType.isXml();
 			case IMPLICIT_XML -> jdbcType.isImplicitXml();
+			case ENUM -> jdbcType.isEnum();
```

Comment:
Please add `Type javaType` as argument to this method and check like this.

```suggestion
			case ENUM -> javaType instanceof Class<?> clz && clz.isEnum();
```